gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,396 @@
+    it can also be constructed using just the summary.    
+    """
+
+    class PlotType(Enum):
+        """Used to select how the metric is to be plotted. Also decides the format of the summary.
+        Example: Summaries of box plots store quartiles, and histogram plotted metrics store a histogram.
+        """
+        BAR = 1         # For scalars
+        BOX = 2         # For 1D distributions
+        HISTOGRAM = 3   # For 1D distributions
+
+    def __init__(
+        self,
+        name: str,
+        data: Optional[Distribution1D] = None,
+        summary: Optional[Dict[str, Any]] = None,
+        store_full_data: bool = True,
+        plot_type: PlotType = None,
+    ) -> GtsfmMetric:
+    """Creates a GtsfmMetric. 
+       Args: 
+            name: name of the metric
+            data: All values of the metric, optional, uses summary if not provided.
+            summary: A summary dict of the metric, generated previously using the same class. 
+                     Has to be provided if data = None.
+            store_full_data: Whether all the values have to be stored and saved or only summary is required. True by default.
+            plot_type: The plot to use for visualization of the metric. 
+                       Defaults:
+                          PlotType.BAR if data is a scalar
+                          PlotType.BOX if data is a distribution (other option is PlotType.HISTOGRAM)
+                        It is inferred from the summary if plot_type is not provided and summary is.
+    """
+        if summary is None and data is None:
+            raise ValueError("Data and summary cannot both be None.")
+
+        self._name = name
+        if data is not None:
+            # Cast to a numpy array
+            if not isinstance(data, np.ndarray):
+                data = np.array(data)
+            if data.ndim > 1:
+                raise ValueError("Metrics must be scalars on 1D-distributions.")
+
+            # Save dimension and plot_type for data
+            self._dim = data.ndim
+            plot_types_for_dim = self._get_plot_types_for_dim(self._dim)
+            if plot_type is None:
+                if summary is not None:
+                    self._plot_type = self.PlotType.HISTOGRAM if "histogram" in summary else self.PlotType.BOX
+                else:
+                    self._plot_type = plot_types_for_dim[0]
+            elif plot_type in plot_types_for_dim:
+                self._plot_type = plot_type
+            else:
+                raise ValueError("Unsupported plot type for the data dimension")
+
+            # Create a summary if the data is a 1D distribution
+            if self._dim == 1:
+                self._summary = self._create_summary(data)
+
+            # Store full data only if its a scalar or if asked to.
+            if self._dim == 0 or store_full_data:
+                self._data = data
+            else:
+                self._data = None
+        else:
+            self._dim = 1
```

Comment:
if data is None, why can't dim be 0-d or 1-d? why only 1d? inline comment would help

Reply:
There is no concept of summary for 0-d. Please refer to the class documentation. Adding a comment nevertheless.

Reply:
could you add that as an inline comment again in line 119, just for clarity again?

Reply:
I added a comment where we set self._dim
# Metrics created from summary alone are 1D distribution metrics
Not sure if you saw this...

Reply:
please excuse the large font before the edit, that was markdown :/